lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1568,7 +1579,17 @@ void IRGenDebugInfoImpl::setCurrentLoc(IRBuilder &Builder,
 
   SILLocation::DebugLoc L;
   SILFunction *Fn = DS->getInlinedFunction();
-  if (Fn && (Fn->isThunk() || Fn->isTransparent())) {
+  if (Opts.DebugInfoFormat == IRGenDebugInfoFormat::CodeView &&
+      Loc.is<ArtificialUnreachableLocation>()) {
+    // There are cases where the ``unreachable`` instruction has an artificial
+    // location, but instructions before and after it point to the same lines.
+    // CodeView gets confused and cannot decide which set of instructions to
+    // break on. It works better if we set this location to be the same as the
+    // one instruction before.
+    // FIXME: I'm guessing this could cause problems in other cases. This should
+    // be tested.
```

Comment:
Can you post a SIL example where this is the case?

Reply:
Oh, I've been looking at the llvm-ir. In the SIL the ``unreachable`` instruction has no line location. Maybe that's why it has an artificial location. I'll post an example.

Reply:
myDivide.swift
``func myDivide(_ a: Int, _ b: Int) -> Int {return a / b}``

myDivide.sil
```
// function_ref _fatalErrorMessage(_:_:file:line:flags:)
%51 = function_ref @$Ss18_fatalErrorMessage__4file4line5flagss5NeverOs12StaticStringV_A2HSus6UInt32VtF : $@convention(thin) (StaticString, StaticString, StaticString, UInt, UI    nt32) -> Never, loc "myDivide.swift":1:52, scope 3 // user: %52
%52 = apply %51(%43, %21, %30, %34, %48) : $@convention(thin) (StaticString, StaticString, StaticString, UInt, UInt32) -> Never, loc "myDivide.swift":1:52, scope 3
unreachable , scope 3                           // id: %53
```

myDivide.ll
```
(instructions relating to a / b)
; <label>:23:                                     ; preds = %22
call swiftcc void@"$Ss18_fatalErrorMessage__4file4line5flagss5NeverOs12StaticStringV_A2HSus6UInt32VtF"(...), !dbg !39
unreachable, !dbg !41
(instructions relating to a / b)
...
!39 = !DILocation(line: 1, column: 52, scope: !40)
!41 = !DILocation(line: 0, scope: !40)
```